spec/beaker/hypervisor/fusion_spec.rb

```diff
@@ -36,6 +36,45 @@ module Beaker
       expect{ Beaker::Fusion.new( @hosts, make_opts) }.to raise_error(/specify a snapshot/)
     end
 
+    context 'connection can be made by host.name' do
+      before :each do
+        allow( fusion ).to receive( :try_ssh_connection )
+      end
+
+      it 'does not set host[:ip] if a connection can be made my host.name' do
+        fusion.provision
+        @hosts.each do |host|
+          expect(host['ip']).to eql(nil)
+        end
+      end
+    end
+
+    context 'if connection cannot be made by host.name,' do
+      before :each do
+        allow( fusion ).to receive( :try_ssh_connection ).and_raise(SocketError)
+      end
+
+      it 'then intropects IP address and sets host[:ip]' do
+        fusion.provision
+        @hosts.each do |host|
+          expect(host['ip']).not_to eql(nil)
+          expect(host['ip']).to eql(MockFission.all.data.find{|h| h.name == host.name}.network_info.data.first[1]['ip_address'])
+        end
+      end
+
+      it 'then host fails init if ip cannot be introspected' do
+        allow_any_instance_of( MockFissionVM ).to receive( :network_info ).and_return( Response.new(0,'',{'eth0' => {'ip_address' => nil}}))
+        expect{ fusion.provision }.to raise_error(/ip address is unavailable/)
+      end
+
+      it 'then sets host dns_name to instance.dns_name' do
+        @hosts.each do |host|
+          expect(fusion).to receive(:set_etc_hosts).with(host, kind_of(String)).exactly(@hosts.count).times
```

Comment:
@anodelman I am not sure about this line. I had to set the receive count to the number of hosts to get it to pass, but my expectation would be that this method would be passed only the host in question.